### PR TITLE
Update entities

### PR DESCRIPTION
**Pull request by AdeHarrison**
# Motivation and Context
Action Rules should exclude receipted Cases.

# What has changed
Entities updated in line with census-rm-case-processor
Integration test data generated using Easy Random

# How to test?
[trello](https://trello.com/c/GdA7flOq/788-rmc-152a-receipting-should-update-case-and-uac-active-flag-8)
[test as part of ](https://github.com/ONSdigital/census-rm-action-scheduler/pull/14)

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[RMC-152(A) Receipting should update case and UAC active flag (8)](https://trello.com/c/GdA7flOq/788-rmc-152a-receipting-should-update-case-and-uac-active-flag-8)

**Review by NickGrantX — CHANGES_REQUESTED** [review]
`Case.receiptReceived` and `UacQidLink.active` have not been replicated - the entity models are drifting apart.

Perhaps it's time to consider that it was a mistake to have `census-rm-case-api` in a different project from `census-rm-case-processor` and go for the monorepo (with two Maven submodules) as was suggested, because there's a constant risk that the JPA entities will drift out of sync with each other, as this PR has shown.

File: src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java [hunk 1]
```diff
@@ -90,7 +91,9 @@
   @Enumerated(EnumType.STRING)
   private CaseState state;
 
-  @OneToMany(mappedBy = "caze")
+  @OneToMany(
+      mappedBy = "caze",
+      cascade = {CascadeType.ALL})
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java:96]: I can understand that this cascade makes it easier to use the `.deleteAll()` convenience method on the repo, but I worry about having the entity classes differ from the ones in `census-rm-case-processor` given that we're sharing the underlying schema. This cascade is only here for testing purposes - is there another way to clear out the test data or do we not have the repos so we can clear out each table individually in the correct sequence?

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Still needs work - looks like the entity POJOs are getting more and more different from the ones in Case Processor, which is going to make keeping them in sync a pain

**Review by NickGrantX — APPROVED** [review]

**Comment by NickGrantX** [comment]
We should put the `cascade` settings into the entities in Case Processor so that they match 1:1

**Review by rjweeks — APPROVED** [review]

**Pull request merged**